lib/fixed_average_info.rb

```diff
@@ -0,0 +1,61 @@
+class FixedAverageInfo
+  Y_SIZES = (1..300).map { |i| 1.2 ** i * 0.000001 }
+
+  attr_reader :data
+
+  def initialize(data)
+    @data = data
+  end
+
+  def as_json(options=nil)
+    averages
+  end
+
+  private
+
+  def averages
+    Session
+    .select(
+      "id, " +
+      "ROUND(longitude / #{grid_x}, 0) AS middle_x, " +
+      "ROUND(latitude / #{grid_y}, 0) AS middle_y "
+    )
+    .find(data[:session_ids])
+    .group_by { |s| [s[:middle_x], s[:middle_y]] }
+    .map do |middle_xy, sessions|
+      {
+        :value => calculate_average(sessions).to_f,
+        :west  =>  middle_xy.first.to_f * grid_x - grid_x / 2,
+        :east  =>  middle_xy.first.to_f * grid_x + grid_x / 2,
+        :south  =>  middle_xy.last.to_f * grid_y - grid_y / 2,
+        :north  =>  middle_xy.last.to_f * grid_y + grid_y / 2
+      }
+    end
+  end
+
+  def calculate_average(sessions)
+    sessions
+    .map { |session| last_hour_average_for(session, data[:sensor_name]) }
+    .sum / sessions.size
+  end
+
+  def last_hour_average_for(session, sensor_name)
+    stream = Stream.where(sensor_name: sensor_name, session_id: session.id).first
+
+    last_measurement_time = stream.measurements.last.time
```

Comment:
turns out we have a default scope for measurements that sorts them by time: :asc, so I don't need to worry about this here. 
On the other hand, maybe we should remove this default scope, cause it probably slows down all the queries and it's not needed in some of them...

Reply:
I'm reaserching that and I think we should at least have an index on Measurement.time since we have this `42   default_scope { order("time ASC") }`in `models/Measurement.rb`. More info: https://www.percona.com/blog/2006/09/01/mysql-order-by-limit-performance-optimization/

Reply:
cc @3v0k4